src/main/java/redis/clients/jedis/JedisSentinelPool.java

```diff
@@ -163,9 +163,7 @@ private HostAndPort initSentinels(Set<String> sentinels, final String masterName
         log.warning("Cannot get master address from sentinel running @ " + hap + ". Reason: " + e
             + ". Trying next one.");
       } finally {
-        if (jedis != null) {
```

Comment:
What if `new Jedis()` eventually throws a runtime exception or someone adds something before this line (https://github.com/xetorthio/jedis/pull/1303/files/ae11c9480520fccbb006890323b55076a526dace#diff-e6525ac1a247b59a49332a6584030684R144) that might raise it?